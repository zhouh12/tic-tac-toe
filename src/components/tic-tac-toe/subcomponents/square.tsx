import React from 'react';
import { observer } from 'mobx-react';
import { SquareValue } from '../../../types/square-value-config';

export interface SquareProps {
  value: SquareValue;
  onClick: () => void;
}

// TODO: in real project, we can create a help function such as bemClassBuilder
// in order to name block or element, right now just hard-coded
const b = 'square';

/**
 * A component fills 'X' or 'O' when clicking it.
 */
export const Square: React.FC<SquareProps> = observer((props) => {
  return (
    <button className={b} onClick={props.onClick}>
      {props.value}
    </button>
  );
});
