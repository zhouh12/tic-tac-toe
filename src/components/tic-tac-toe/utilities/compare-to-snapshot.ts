import { ReactElement } from 'react';
import { create, TestRendererOptions } from 'react-test-renderer';

/**
 * A helper function creates snapshot test
 */
export function compareToSnapshot(
  nextElement: ReactElement<any>,
  options?: TestRendererOptions
) {
  const tree = create(nextElement, options);
  expect(tree).toMatchSnapshot();
  tree.unmount();
}
