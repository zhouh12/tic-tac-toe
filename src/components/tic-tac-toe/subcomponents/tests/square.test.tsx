import { compareToSnapshot } from '../../utilities/compare-to-snapshot';
import React from 'react';
import { Square, SquareProps } from '../square';

describe('Square', () => {
  const onClickFunc = jest.fn();
  const testProps: SquareProps = {
    value: 'X',
    onClick: onClickFunc,
  };

  it('renders correctly', () => {
    compareToSnapshot(<Square {...testProps} />);
  });
});
