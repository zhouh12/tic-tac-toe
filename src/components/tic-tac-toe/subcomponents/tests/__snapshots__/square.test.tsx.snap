// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square renders correctly 1`] = `
<button
  className="square"
  onClick={[MockFunction]}
>
  X
</button>
`;
